### DISCUSSION TITLE
Getx 4.2.0

### DISCUSSION THREAD

**Discussion by jonataslaw** [jonataslaw/getx#1649] [General]
## [4.2.0] - Big update

This update fixes important bugs as well as integrates with Navigator 2. It also adds GetRouterOutlet, similar to angular RouterOutlet thanks to @ahmednfwela. Also, the documentation translation for Vietnamese (@khangahs) has been added, making the GetX documentation available for 11 different languages, which is just fantastic for any opensource project. GetX has achieved more than 5.4k likes from the pub, and more than 4k stars on github, has videos about it with 48k on youtube, and has communities in the 4 hemispheres of the earth, besides having a large list of contributors as you see bellow. We're all happy to facilitate development with dart and flutter, and that making programming hassle-free has been taken around the world.

Changes in this version:

- Fix: Navigating to the same page with Get.offNamed does not delete the controller from that page using Get.lazyPut.

- Fix Readme GetMiddleware typos 
by @nivisi

- Fix url replace error
by @KevinZhang19870314 

- Changed response default encoding from latin1 to utf8 
by @heftekharm

- Add Duration in ExtensionBottomSheet
by @chanonpingpong

- Added compatibility with dart-lang/mockito
by @lifez

- Added extensions methods to convert value in percent value 
by @kauemurakami

- Set darkTheme equal theme when darkTheme is null 
by @eduardoFlorence

- Add padding to 'defaultDialog' 
by @KevinZhang19870314

- GraphQLResponse inherit Response info 
by @jasonlaw

- Fix Redundant concatenating base url 
by @jasonlaw

- Add content type and length into the headers when the content type is 'application/x-www-form-urlencoded'
by @calvingit

- Make withCredentials configurable 
by @jasonlaw

- Fix flutter 2.0 error
by @yunchiri

- Allow deleting all registered instances
by @lemps

- Refactor/rx interface notify children 
@by kranfix

- Fixed parameter parsing and middleware sorting 
by @ahmednfwela

- Improvements to router outlet 
by @ahmednfwela

- Minor improvements and bug fixes 
by @ahmednfwela

- Adding route guards and improving navigation 
by @ahmednfwela

- Fix RxInterface.proxy losing its previous value on exception
by @WillowWisp

- Added dispose() for bottomSheet.
by @furkankurt 

- Added Pull request template 
by @unacorbatanegra

- Fix and update documentation:
@Farid566, 
@galaxykhh, 
@arslee07, 
@GoStaRoff, 
@BondarenkoArtur, 
@denisrudnei,
@Charly6596, 
@nateshmbhat, 
@hrithikrtiwari, 
@Undeadlol1,
@rws08,
@inuyashaaa,
@broccolism, 
@aadarshadhakalg,
@ZeroMinJeon

<hr /><em>This discussion was created from the release <a href='https://github.com/jonataslaw/getx/releases/tag/4.2.0'>Getx 4.2.0</a>.</em>

**Comment by gairick-saha** [comment]
Please provide a documentation for navigator 2.0

**Reply by kechkibet** [reply]
+1

**Comment by NiranjanShah** [comment]
+1

Regards,

Niiranjan G Mandan
